packages/epics/src/agreements/plugins/components/common/token-payout-field.tsx

```diff
@@ -42,20 +42,19 @@ export const TokenPayoutField = ({
       onChange({ amount: '', token: value.token });
       return;
     }
-    const parsed = Number.parseInt(amount, 10);
-    onChange({
-      amount: String(Number.isNaN(parsed) ? 0 : parsed),
-      token: value.token,
-    });
+    if (/^\d*(?:[\.\,]\d*)?$/.test(amount) || amount === '') {
+      onChange({ amount, token: value.token });
```

Comment:
Regex returns true for an empty string, so the check is redundant
```suggestion
    if (/^\d*(?:[\.\,]\d*)?$/.test(amount)) {
      onChange({ amount, token: value.token });
```

Reply:
This might make sence.